lint/lostSQLCloseForRows_checker.go

```diff
@@ -0,0 +1,140 @@
+package lint
+
+//! Detects use *sql.Rows without call Close method.
+//
+// @Before:
+// rows, _ := db.Query(...)
+// for rows.Next {
+//   ...
+// }
+//
+// @After:
+// rows, _ := db.Query(...)
+// for rows.Next {
+//   ...
+// }
+// rows.Close
+
+import (
+	"go/ast"
+	"go/types"
+)
+
+func init() {
+	addChecker(&lostSQLCloseForRowsChecker{}, attrExperimental)
+}
+
+type lostSQLCloseForRowsChecker struct {
+	checkerBase
+}
+
+// Warning if sql.Rows local variables (including function parameters):
+// 1. Not using as parameter in other functions call;
+// 2. Not returning in functions results;
+// 3. Not call Close method for variable;
+
+func (c *lostSQLCloseForRowsChecker) VisitFuncDecl(decl *ast.FuncDecl) {
+	// Function parameter variable
+	params := decl.Type.Params
+	var paramVal types.Object
+	for _, p := range params.List {
+		t := c.ctx.typesInfo.TypeOf(p.Type)
+		if t != nil {
+			switch t.String() {
```

Comment:
Can be in if above, so `if t != nil && t.String() == "*database/sql.Rows" {`

Reply:
Ok. I fix it...